OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -125,6 +153,88 @@ public boolean onItemLongClick(AdapterView<?> parent, View view, int position, l
         return view;
     }
 
+    private void toggleCourseOutlineDowloadFooter(final CourseComponent courseComponent) {
+        if (isOnCourseOutline) {
+            final int totalDownloadableVideos = courseComponent.getDownloadableVideosCount();
+            // support video download for video type excluding the ones only viewable on web
+            if (totalDownloadableVideos > 0) {
+                int downloadedCount = environment.getDatabase().getDownloadedVideosCountForCourse(courseData.getCourse().getId());
+
+                if (downloadedCount == totalDownloadableVideos) {
+                    Long downloadTimeStamp = environment.getDatabase().getLastVideoDownloadTimeForCourse(courseData.getCourse().getId());
```

Comment:
How are we handling the edge cases here, what if function will return different value than expected like 0 or -1?

Reply:
The current logic meets these scenarios as well. If `downloadedCount` is 0 or -1 then it will show `Download content` footer. 
If `totalDownloadableVideos` is 0 or -1 then, it won't show the footer at all.